docs/frontend-system/building-apps/08-migrating.md

```diff
@@ -0,0 +1,435 @@
+const legacyFeatures = convertLegacyApp(
+  <>
+    <AlertDisplay transientTimeoutMs={2500} />
+    <OAuthRequestDialog />
+    <AppRouter>
+      <Root>{routes}</Root>
+    </AppRouter>
+  </>,
+);
+
+const app = createApp({
+  /* other options */
+  features: [...legacyFeatures],
+});
+
+export default app.createRoot();
+```
+
+We've taken all the elements that were previously passed to `app.createRoot(...)`, and instead passed them to `convertLegacyApp(...)`. We then pass the features returned by `convertLegacyApp` and forward them to the `features` option of the new `createApp`.
+
+There is one more details that we need to deal with before moving on. The `app.createRoot()` function now returns a React element rather and a component, so we need to update our app `index.tsx` as follows:
+
+```tsx title="in packages/app/src/index.tsx"
+import '@backstage/cli/asset-types';
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+// highlight-remove-next-line
+import App from './App';
+// highlight-add-next-line
+import app from './App';
+
+// highlight-remove-next-line
+ReactDOM.createRoot(document.getElementById('root')!).render(<App />);
+// highlight-add-next-line
+ReactDOM.createRoot(document.getElementById('root')!).render(app);
+```
+
+At this point the contents of your app should be past the initial migration stage, and we can move on to migrating any remaining options that you may have passed to `createApp`.
+
+## Migrating `createApp` Options
+
+Many of the `createApp` options have been migrated to use extensions instead. Each will have their own [extension creator](../architecture/03-extensions.md#extension-creators) that you use to create a custom extension. To add these standalone extensions to the app they need to be passed to `createExtensionOverrides`, which bundles them into a _feature_ that you can install in the app. See the [standalone extensions](../architecture/05-extension-overrides.md#create-standalone-extensions) section for more information.
+
+For example, assuming you have a `lightTheme` extension that you want to add to your app, you can use the following:
+
+```ts
+const app = createApp({
+  features: [
+    createExtensionOverrides({
+      extensions: [lightTheme],
+    }),
+  ],
+});
+```
+
+### `apis`
+
+[Utility API](../utility-apis/01-index.md) factories are now installed as extensions instead. Pass the existing factory to `createApiExtension` and install it in the app. For more information, see the section on [configuring Utility APIs](../utility-apis/04-configuring.md).
+
+For example, the following API configuration:
+
+```ts
+const app = createApp({
+  apis: [
+    createApiFactory({
+      api: scmIntegrationsApiRef,
+      deps: { configApi: configApiRef },
+      factory: ({ configApi }) => ScmIntegrationsApi.fromConfig(configApi),
+    }),
+  ],
+});
+```
+
+Can be converted to the following extension:
+
+```ts
+const scmIntegrationsApi = createApiExtension({
```

Comment:
It wasn't immediate obvious to me but do we simply pass these extension creator return values mentioned throughout this doc into `createApp` the same way we provide the `lightTheme` extension mentioned above?
```tsx
 const app = createApp({
  features: [
    createExtensionOverrides({
      extensions: [lightTheme, scmIntegrationsApi, signInPage, ...],
    }),
  ],
});
```

Reply:
Yep! I added in an extra sentence after the `lightTheme` example to try to clarify it a bit more. I did want to try to avoid having to repeat the `createExtensionOverrides` throughout the entire doc, especially consider that you probably wanna group things together too 😅